cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/dataset/BusinessMetadataDatasetTest.java

```diff
@@ -43,7 +44,13 @@
   private static final Id.DatasetInstance datasetInstance =
     Id.DatasetInstance.from(DatasetFrameworkTestUtil.NAMESPACE_ID, "meta");
 
-  BusinessMetadataDataset dataset;
+  private BusinessMetadataDataset dataset;
+
+  private final Id.Application app1 = Id.Application.from("ns1", "app1");
+  // Have to use Id.Program here because Id.Flow does not implement equals. It needs to be updated.
```

Comment:
Would it be a good idea to make Id.Program#equals check if the other object is an instance of Id.Program, instead of equals that class? And then the type can be checked also?
Currently, the equals doesn't check type, which is not good.